lib/grandpa/message_handler_test.go

```diff
@@ -177,6 +177,57 @@ func TestMessageHandler_FinalizationMessage_NoCatchUpRequest(t *testing.T) {
 	cm, err := fm.ToConsensusMessage()
 	require.NoError(t, err)
 
+	h := NewMessageHandler(gs, st.Block)
+	out, err := h.HandleMessage(cm)
+	require.EqualError(t, err, "signature is not valid")
+	require.Nil(t, out)
+
+}
+
+func TestMessageHandler_FinalizationMessage_NoCatchUpRequest_ValidSig(t *testing.T) {
+	st := newTestState(t)
+	voters := newTestVoters(t)
```

Comment:
hm, something in this test seems off, `newTestVoters` should return 9 votes, so a finalization message with only 1 signature shouldn't meet the threshold, maybe I am missing something?

Reply:
No, I fixed verify so this test fails, going to add tests...